eclipse-theia/theia
# vsx-registry: Skip extension resolution if already installed

#### What it does

Closes #10560 by comparing the versions of already installed extensions with the to-be-resolved extension version. When the extension already exists and its version is greater or equal to the one that should be installed, its download is skipped.

#### How to test

1. Install the `hbenl.test-adapter-converter` and `hbenl.vscode-test-explorer` (in this exact order) extension from [here](https://github.com/eclipse-theia/theia/files/7856099/hbenl.test-extensions.zip). The `hbenl.vscode-test-explorer` version has been increased to `2.42.0` to recreate the original issue.
2. Search for the "Mocha Test Explorer" (`hbenl.vscode-mocha-test-adapter`) and install it. It requires the `hbenl.version-test-explorer` as an extension dependency.
3. In the console where Theia was started, the message `[hbenl.vscode-test-explorer]: is already installed with the same or newer version '2.42.0'` should appear.
4. Confirm that no old version of the test adapter extension has been downloaded into your `<user>/.theia` directory.

---

5. Repeat all of the above but instead of installing the extension from the `.vsix` file, unpack the extensions into your `plugins` folder. Additionally perform the test by adding the `.vsix` directly to the `plugins` folder without unpacking them.

---

6. Restart the application and also confirm here that no downgrade is performed. 

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
